### PR TITLE
Add SkipList Data Structure

### PR DESCRIPTION
**Pull request by TawfikYasser**
* Adding the skiplist data structure.

### CODE REVIEW

File: Lecture-16-Skip-List/SkipList.cpp [hunk 1]
```diff
@@ -0,0 +1,152 @@
+#include <iostream>
+#include <cstdlib>
+#include <cmath>
+#include <cstring>
+#include<time.h>
+#define MAX_LEVEL 6
```
Comment (cs-MohamedAyman) [inline Lecture-16-Skip-List/SkipList.cpp:6]: Please remove this macro, I prefer to build the skip list dynamically.

File: Lecture-16-Skip-List/SkipList.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        header = new snode(MAX_LEVEL, value);
+        level = 0;
+    }
+    ~skiplist()
+    {
+        delete header;
+    }
+    void print_list();
+    bool search_element(int &);
+    void insert_element(int &);
```
Comment (cs-MohamedAyman) [inline Lecture-16-Skip-List/SkipList.cpp:45]: There is a missing mandatory feature "delete_element()".

File: Lecture-16-Skip-List/SkipList.cpp (hunk 1 above)
Comment (TawfikYasser) [reply Lecture-16-Skip-List/SkipList.cpp:6]: Do you mean that we should takes MAX_LEVEL as an input from the user?

File: Lecture-16-Skip-List/SkipList.cpp (hunk 2 above)
Comment (TawfikYasser) [reply Lecture-16-Skip-List/SkipList.cpp:45]: Ok, I will add the delete feature.

File: Lecture-16-Skip-List/SkipList.cpp (hunk 1 above)
Comment (cs-MohamedAyman) [reply Lecture-16-Skip-List/SkipList.cpp:6]: I prefer to increase and decrease it based on the current number of elements in the data structure.

Comment (TawfikYasser) [reply Lecture-16-Skip-List/SkipList.cpp:6]: I have added the delete feature, For MAX_LEVEL i will update it soon and i will make another PR, For now i suggest to make MAX_LEVEL as is, May any one make it. Okay?

File: Lecture-16-Skip-List/SkipList.cpp [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            update[i]->forw[i] = x->forw[i];
+        }
+        delete x;
+        while (level > 0 && header->forw[level] == NULL)
+        {
+            level--;
+        }
+    }
+    cout<<value<<" deleted successfully!"<<endl;
+}
```
Comment (TawfikYasser) [inline Lecture-16-Skip-List/SkipList.cpp:151]: Delete Function

**Pull request closed without merging**